pkg/loadbalancer/smooth_weighted_round_robin_test.go

```diff
@@ -0,0 +1,310 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package loadbalancer
+
+import (
+	cryptoRand "crypto/rand"
+	"math"
+	"math/big"
+	"math/rand"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type server struct {
+	name   string
+	weight float64
+}
+
+var _ = Describe("Smooth Weighted RR", func() {
+	// Global Vars
+	var servers []server
+	var smoothTestingServers []server
+	var roundRobinServers []server
+	var lb *smoothWeightedRR
+	// Helpers
+	randFloat := func() float64 {
+		minWeight := 0.0
+		maxWeight := 20.0
+		rInt, err := cryptoRand.Int(cryptoRand.Reader, big.NewInt(math.MaxInt64))
+		if err != nil {
+			return 10.0
+		}
+		f := float64(rInt.Int64()) / (1 << 63)
+		return minWeight + f*(maxWeight-minWeight)
+	}
+
+	addServer := func(index int, lb *smoothWeightedRR) server {
+		server := servers[index]
+		err := lb.Add(server.name, server.weight)
+		if err != nil {
+			Fail(err.Error())
+		}
+		return server
+	}
+
+	addAllServers := func(lb *smoothWeightedRR) {
+		for i := 0; i < len(servers); i++ {
+			_ = addServer(i, lb)
+		}
+	}
+
+	getTotalServesWeight := func(servers []server) float64 {
+		var weight float64
+		for _, v := range servers {
+			weight += v.weight
+		}
+		return weight
+	}
+
+	// Validations
+	validateServerAdded := func(index int, lb *smoothWeightedRR) {
+		server := servers[index]
+		serversMap := lb.itemsMap
+		Expect(serversMap[server.name].weight).To(Equal(server.weight))
+	}
+
+	validateAllServersAdded := func(lb *smoothWeightedRR) {
+		for i := 0; i < len(servers); i++ {
+			validateServerAdded(i, lb)
+		}
+	}
+
+	validateEmptyLBState := func(lb *smoothWeightedRR) {
+		Expect(lb.ItemsCount()).To(Equal((0)))
+		for i := 0; i < 100; i++ {
+			Expect(lb.Next()).To(BeNil())
+		}
+	}
+
+	validateLoabBalancing := func(rounds int, servers []server, lb *smoothWeightedRR) {
```

Comment:
```suggestion
	validateLoadBalancing := func(rounds int, servers []server, lb *smoothWeightedRR) {
```

Reply:
Am I missing something here or there is no difference in the suggestion?

Reply:
sic - 'b' -> 'd'